### PR TITLE
Help Commands

### PR DESCRIPTION
**Pull request by nca6109**
[Ticket](https://github.com/orgs/RIT-Crowd-Comic/projects/2/views/1?pane=issue&itemId=63986884)

### Description
- Use `/help` to be sent a message with an embed of available commands
- Use '/help-remember' to be sent a message with an embed of all available remember related commands

#### Parameters
N/A

### Notes
Following the refactor in sprint 2, I will make the command lists fill dynamically. Additionally, this system would work best if every command were adequately marked as devOnly or not, otherwise dev commands will end up in the '/help' list.

### CODE REVIEW

File: src/commands/misc/help.js
Comment (BlckHawker) [inline src/commands/misc/help.js:7]: fix spelling mistake: interation -> interaction

File: src/commands/misc/helpRemember.js
Comment (BlckHawker) [inline src/commands/misc/helpRemember.js:12]: Is it possible to add fields dynamically so the name is the name of the command and the value is the description? I feel it would be easier as we add or remove commands from the bot

File: src/commands/misc/helpRemember.js
Comment (BlckHawker) [inline src/commands/misc/helpRemember.js:4]: If we are planning to add subcommands later, would it be better for this to be a subcommand itself rather than its own command?

File: src/commands/misc/help.js
Comment (BlckHawker) [inline src/commands/misc/help.js:14]: I'm usnure if any of the fields here are neccessary besides '/help-remember' as the rest are debug commands. You can add another field for reminder features as those aren't stated anywhere

**Review by BlckHawker — CHANGES_REQUESTED** [review]
## PR Description Review
- I recommend writing detailed PR descriptions so people don't have to look through each file to understand how the commands work. You can look at other PRs for examples.
    - This will also speed up documentation writing next week as you can copy and paste what you write here.
    - This can also help catch if the intention of the command and the inplmentation are different

Here is some data to have in your description: 
- the name of the command 
- any paramters including name, description and if they're optional
    - If they're optional, state what the default value is if not provded
- Screenshots of each command being used
- Any notes or warnings that you think the team should make of note of. i.e If you think this feature will break once a future feauture is inplemented, and how we could avoid that. 


## PR Review 
It looks good to start, but I think more work should be added before it's merged. It's probably best to wait until other PRs are merged if we are adding fields by hand. Otherwise, I suggest finding a way to do them dynamically. This could be similar to how the commands are registered to the bot.

Only main request is to add the check in commands to the help command

File: src/commands/misc/helpRemember.js
Comment (nca6109) [reply src/commands/misc/helpRemember.js:4]: Most likely, yes

File: src/commands/misc/helpRemember.js
Comment (tstodter) [reply src/commands/misc/helpRemember.js:12]: Agreed that we should figure a way to do this dynamically, or at least by keeping the name/description with each command and pulling it in here (so we're not duplicating that data in the repo).

Comment (tstodter) [reply src/commands/misc/helpRemember.js:12]: For now I suggest leaving it as it is.

File: src/commands/misc/help.js
Comment (tstodter) [inline src/commands/misc/help.js:11]: Indentation is a touch odd here, recommend
```suggestion
        const embed = new EmbedBuilder()
            .setTitle('Server Commands')
            .addFields(
```

File: src/commands/misc/helpRemember.js
Comment (wct9667) [inline src/commands/misc/helpRemember.js:45]: typo: `interation` should be `interaction`. Help command throws an error.

**Review by wct9667 — CHANGES_REQUESTED** [review]

**Review by wct9667 — APPROVED** [review]

**Review by BlckHawker — APPROVED** [review]
sonic-boom-knuckles

**Pull request merged**
